src/urllib3/exceptions.pyi

```diff
@@ -0,0 +1,50 @@
+from typing import Any, Optional, Union, Tuple
+
+class HTTPError(Exception): ...
+class HTTPWarning(Warning): ...
+
+class PoolError(HTTPError):
+    pool: Any
+    def __init__(self, pool: Any, message: str) -> None: ...
+    def __reduce__(self) -> Union[str, Tuple[Any, ...]]: ...
+
+class RequestError(PoolError):
+    url: Any
+    def __init__(self, pool: Any, url: str, message: str) -> None: ...
+    def __reduce__(self) -> Union[str, Tuple[Any, ...]]: ...
+
+class SSLError(HTTPError): ...
+class ProxyError(HTTPError): ...
+class DecodeError(HTTPError): ...
+class ProtocolError(HTTPError): ...
+
+ConnectionError: Any
+
+class MaxRetryError(RequestError):
+    reason: Any
+    def __init__(self, pool: Any, url: str, reason=Optional[str]) -> None: ...  # type: ignore
+
+class HostChangedError(RequestError):
+    retries: Any
+    def __init__(self, pool: Any, url: str, retries=int) -> None: ...  # type: ignore
```

Comment:
I'm a little puzzled why I keep getting a mypy error here despite adding more types, so included an ignore.

```
src/urllib3/exceptions.pyi:25: error: Function is missing a type annotation for one or more arguments
src/urllib3/exceptions.pyi:29: error: Function is missing a type annotation for one or more arguments.
```

Reply:
Should be `retries: int` instead of =int :)

Reply:
LOL I guess this is where I should have taken a break :D